src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/PostgresDirectoryDAO.scala

```diff
@@ -449,18 +449,28 @@ class PostgresDirectoryDAO(protected val writeDbRef: DbReference, protected val
     serializableWriteTransaction("setUserAzureB2CId", samRequestContext) { implicit session =>
       val u = UserTable.column
       val results =
-        samsql"""update ${UserTable.table}
+        if (b2cId.value == "")
```

Comment:
In the data-access layer, I think it would be better to accept a real value or `null`. The service layer, or even the API layer, should be the one to translate `""` to `null`.